### PR TITLE
Do not fail parsing on EdDSA signatures that are not 64 bytes

### PR DESCRIPTION
**Pull request by zapu**
Fixes https://github.com/keybase/keybase-issues/issues/2787

We will now parse such signatures correctly, but we might not be able to verify them. If signature buffer is too short, we will just stack zeroes at the end, so `nacl` does not throw.

### CODE REVIEW

File: test/files/eddsa.iced [hunk 1]
```diff
@@ -119,3 +119,84 @@ exports.sign_and_exchange = (T, cb) ->
...
+kGOZKibkwoGC8irPGmBzN6tAf+ravK+HAPg7A3NA+Yqr+aTzl2xSXqIAOb2fJjIA
+AC0Xg++AATao
+=uJ/2
+-----END PGP PUBLIC KEY BLOCK-----
+"""
+
+exports.invalid_userid_sig = (T, cb) ->
+  await KeyManager.import_from_armored_pgp { armored: key_second_userid_invalid }, defer err, keym
+  T.no_error err # imported fine
+  T.assert keym.userids.length == 1 # but only one userid (of two)
```
Comment (zapu) [inline test/files/eddsa.iced]: It should also find this other userid and verify that it indeed failed to verify, but I didn't know how

Comment (maxtaco) [inline test/files/eddsa.iced]: Can you just verify that the userid that popped out is the right user ID?  I.e., assert that `keym.userids[0]` matches some pattern?

File: src/openpgp/util.iced [hunk 2]
```diff
@@ -53,3 +53,16 @@ exports.ops_to_keyflags = ops_to_keyflags = (ops) ->
...
 
 #=========================================================
+
+# Fits a buffer to exactly `size` bytes, by either padding it with
+# zeroes or trimming it when needed. The example use of this would be
+# a malformed signature, we'd rather fit our buffers and fail to
+# verify, than refuse parsing altogether.
+exports.fit_to_size = fit_to_size = (size, buf) ->
+  l = size - buf.length
+  if l == 0
```
Comment (maxtaco) [inline src/openpgp/util.iced]: use `is`

File: test/files/eddsa.iced [hunk 3]
```diff
@@ -71,7 +71,7 @@ exports.import_key_1 = (T,cb) ->
 exports.verify_sigs = (T,cb) ->
   for sig,i in sigs
     await unbox { armored : sig, keyfetch : km  }, defer err, literals, warnings
-    T.no_error err, "no errors pls #{i}"
+    T.no_error err, "no errors"
```
Comment (maxtaco) [inline test/files/eddsa.iced]: what's wrong with `#{i}` here?

Comment (zapu) [inline test/files/eddsa.iced]: oh, oops, we can keep the `#{i}`

**Pull request merged**